server/publications/collections/reactiveAggregate.js

```diff
@@ -0,0 +1,102 @@
+import _ from "lodash";
+import { Meteor } from "meteor/meteor";
+import { check, Match } from "meteor/check";
+import { Mongo, MongoInternals } from "meteor/mongo";
+
+
+// Add the aggregate function available in tbe raw collection to normal collections
+Mongo.Collection.prototype.aggregate = function (pipelines, options) {
+  const coll = this._getCollection();
+  return Meteor.wrapAsync(coll.aggregate.bind(coll))(pipelines, options);
+};
+
+Mongo.Collection.prototype._getDb = function () {
+  if (typeof this._collection._getDb === "function") {
+    return this._collection._getDb();
+  }
+  const  mongoConn = MongoInternals.defaultRemoteCollectionDriver().mongo;
+  return wrapWithDb(mongoConn);
+};
+
+Mongo.Collection.prototype._getCollection = function () {
+  const db = this._getDb();
+  return db.collection(this._name);
+};
+
+function wrapWithDb(mongoConn) {
+  if (mongoConn.db) {
+    return mongoConn.db;
+  }
+}
+
+/**
+ * Create a Reactive collection from a Mongo aggregate pipeline
+ * @param {Object} sub - The publication we are creating
+ * @param {Collection} collection - the collection we are operating on
+ * @param {Array} pipeline - The mongo aggregation pipeline to run
+ * @param {Object} options - an object of options
+ * @returns {Cursor} A mongo cursor for subscription
+ * @constructor
```

Comment:
Should this be using `@constructor`? My understanding of that tag would be that it's used for classes.
 @machikoyasuda ?